clippy_lints/src/collapsible_if.rs

```diff
@@ -209,6 +211,64 @@ impl CollapsibleIf {
         !matches!(cond.kind, ExprKind::Let(..))
             || (cx.tcx.sess.edition().at_least_rust_2024() && self.msrv.meets(cx, msrvs::LET_CHAINS))
     }
+
+    fn check_collapsible_if_nested_if_else(&self, cx: &LateContext<'_>, if_expr: &Expr<'_>) {
+        if let Some(If {
+            cond,
+            then: Expr {
+                kind: ExprKind::Block(then_block, ..),
+                ..
+            },
+            r#else: Some(else_expr),
+        }) = If::hir(if_expr)
+            && let Some(inner_if_expr) = expr_block(then_block)
+            && let Expr {
+                kind: ExprKind::If(inner_cond, inner_then, ..),
+                ..
+            } = inner_if_expr
+            && !block_starts_with_significant_tokens(cx, then_block, inner_if_expr, self.lint_commented_code)
+            && SpanlessEq::new(cx).eq_expr(inner_then, else_expr)
+        {
+            let mut app = Applicability::MachineApplicable;
+
+            let first_cond_sugg = match cond.kind {
+                ExprKind::Binary(bin_op, lhs, rhs) => {
+                    let new_bin_op = if let BinOpKind::Ne = bin_op.node {
+                        BinOpKind::Eq
+                    } else {
+                        BinOpKind::Ne
+                    };
```

Comment:
What about other binary operations? Right now, the following

```rust
    let a = 10;
    let b = 20;

    if a < 20 {
        if b < 30 {
            println!("Hello");
        }
    } else {
        println!("world!");
    }
```

will be transformed into the (non equivalent)

```rust
    if a != 20 || b < 30 {
        println!("Hello world!");
    }
```

You should look at the `!` operator for `Sugg` and let it invert the operation, it might be beneficial.